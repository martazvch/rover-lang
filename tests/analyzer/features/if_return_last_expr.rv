fn add() -> int {
    return if true do 1 else 5
}

fn add1() -> int {
    return if false { break 1 } else { break 5 }
}

fn add2() -> int {
    var trash = if true do return 1 else 5
    return trash
}

fn add3() -> int {
    var trash = if true do 1 else return 5
    return trash
}

fn add4() -> int {
    if true do return 1 else return 5
}

fn add5() -> int {
    return if true do 1 else 5
}

fn add6() -> int {
    return if true do return 1 else 5
}

fn add7() -> int {
    return if true do 1 else return 5
}

fn add8() -> int {
    return if true do return 1 else return 5
}

fn main() {}
