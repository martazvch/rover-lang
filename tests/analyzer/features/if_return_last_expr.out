[Function declaration add, returns]
    [Return]
        [If]
        - condition:
            [Bool true]
        - then:
            [Int 1]
        - else:
            [Int 5]

[Function declaration add1, returns]
    [Return]
        [If]
        - condition:
            [Bool false]
        - then:
            [Block pop count: 0, is_expr: true]
                [Break depth: 0]
                    [Int 1]
        - else:
            [Block pop count: 0, is_expr: true]
                [Break depth: 0]
                    [Int 5]

[Function declaration add2, returns]
    [Variable declaration index: 0, scope: local]
        [If]
        - condition:
            [Bool true]
        - then:
            [Return]
                [Int 1]
        - else:
            [Int 5]
    [Return]
        [Variable index: 0, scope: local]

[Function declaration add3, returns]
    [Variable declaration index: 0, scope: local]
        [If]
        - condition:
            [Bool true]
        - then:
            [Int 1]
        - else:
            [Return]
                [Int 5]
    [Return]
        [Variable index: 0, scope: local]

[Function declaration add4, returns]
    [If]
    - condition:
        [Bool true]
    - then:
        [Return]
            [Int 1]
    - else:
        [Return]
            [Int 5]

[Function declaration add5, returns]
    [Return]
        [If]
        - condition:
            [Bool true]
        - then:
            [Int 1]
        - else:
            [Int 5]

[Function declaration add6, returns]
    [Return]
        [If]
        - condition:
            [Bool true]
        - then:
            [Return]
                [Int 1]
        - else:
            [Int 5]

[Function declaration add7, returns]
    [Return]
        [If]
        - condition:
            [Bool true]
        - then:
            [Int 1]
        - else:
            [Return]
                [Int 5]

[Function declaration add8, returns]
    [Return]
        [If]
        - condition:
            [Bool true]
        - then:
            [Return]
                [Int 1]
        - else:
            [Return]
                [Int 5]

[Function declaration main]
